### PR TITLE
Added operations to Doubly and Circular Linked List

### PR DESCRIPTION
**Pull request by aayushinigam**

### CODE REVIEW

File: Data Structures/Linked-Lists/Circular Linked List/deletion.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ 
+  int data;
+  node *next;
+};
+
+/* Deletes a node of given address
+    node** - address of the HEAD (beginning of linked list)
+    given_node* - node to be deleted(its address is passed as argument)
+*/
+void del(node **head, node *given_node){
```
Comment (arnav-mandal1234) [inline Data Structures/Linked-Lists/Circular Linked List/deletion.cpp]: proper function name "delete"

File: Data Structures/Linked-Lists/Circular Linked List/deletion.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+   There are 3 cases: 1. node to be deleted is the last node
+                      2. node to be deleted is the only node in list.
+                      3. node to be deleted is present at the beginning of list
+   
+   To run this code, make sure you make appropriate calls to this function in main().
+*/
+
+
+struct node {
+
```
Comment (arnav-mandal1234) [inline Data Structures/Linked-Lists/Circular Linked List/deletion.cpp]: indentation problem.

File: Data Structures/Linked-Lists/Circular Linked List/insertion.cpp [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                           3.Adding the node at last.
+                           4.Adding the node after a given node.
+                           5.Adding the node before a given node.
+
+To run this code, make sure you make approriate calls to the given function in main() 
+
+*/
+
+struct node {
+
```
Comment (arnav-mandal1234) [inline Data Structures/Linked-Lists/Circular Linked List/insertion.cpp]: indentation problem fix these in all files.

File: Data Structures/Linked-Lists/Circular Linked List/deletion.cpp [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      temp->next = given_node->next;
+  }
+ 
+  delete given_node;
+  return;
+ 
+}
+  
+/* Time Complexity : O(n) if node is present at the end or is present between two nodes.
+                     O(1) if node is present at the beginning.
```
Comment (arnav-mandal1234) [inline Data Structures/Linked-Lists/Circular Linked List/deletion.cpp]: Time complexity is a general thing so it should only be O(n). Mention single line explanation for O(n) time complexity.

File: Data Structures/Linked-Lists/Circular Linked List/deletion.cpp [hunk 5]
```diff
@@ -0,0 +1,52 @@
+/* Deletes a node of given address in a circular linked list where last node's next points to first node.
```
Comment (arnav-mandal1234) [inline Data Structures/Linked-Lists/Circular Linked List/deletion.cpp]: Every file should have the main function. you can combine insertion and deletion in one file.

File: Data Structures/Linked-Lists/Circular Linked List/deletion.cpp (hunk 1 above)
Comment (aayushinigam) [reply Data Structures/Linked-Lists/Circular Linked List/deletion.cpp]: "delete" is a keyword in c++, it will case an error if I name it delete.

Comment (arnav-mandal1234) [reply Data Structures/Linked-Lists/Circular Linked List/deletion.cpp]: Okay, then why not deleteNode?
I mean to say that it should have a proper name.

Comment (aayushinigam) [reply Data Structures/Linked-Lists/Circular Linked List/deletion.cpp]: Sure! I'll do that.

File: Data Structures/Linked-Lists/Circular Linked List/insertion.cpp [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+/* Inserts new node after/before a given_node 
+   node** - address of the HEAD (beginning of linked list)
+   node *given_node - node before/after which the new node has to be inserted
+   int x-  value to be added in the linked list (integer in this example)
+   int insertion_position - 0 for insertion before the given_node
+                            1- for insertion after the given_node
+*/
+
+void insert_given_node(node **head,node *given_node, int x,int insertion_position){
```
Comment (aditya81070) [inline Data Structures/Linked-Lists/Circular Linked List/insertion.cpp]: Follow [Coding Standards](https://github.com/sahilbansal17/Competitive_Coding/wiki/Coding-Standards#naming-convention)

File: Data Structures/Linked-Lists/Circular Linked List/insertion.cpp [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+  }
+}
+
+/* Time Complexity - O(1) for 1.adding node when list is empty
+                              2. adding node after a given node.
+                     O(n) for 1. adding node at the beginning of a non-empty list.
+                              2. adding node at the end
+                              3. adding node before a given node.
+*/ 
+
```
Comment (aditya81070) [inline Data Structures/Linked-Lists/Circular Linked List/insertion.cpp]: Remove unnecessary blank lines.

File: Data Structures/Linked-Lists/Circular Linked List/reversal.cpp [hunk 8]
```diff
@@ -0,0 +1,35 @@
+/* Reverses a Doubly Linked List i.e. it swaps prev and next of every single node and 
+  head points to the last node in original list after.-
+
+  To runt this code, mak eyou make appropriate calls to the given function in mian()
```
Comment (aditya81070) [inline Data Structures/Linked-Lists/Circular Linked List/reversal.cpp]: Fix the typos.

**Comment by aditya81070** [comment]
@aayushinigam Please make appropriate changes to your code.

**Comment by aayushinigam** [comment]
Yes I will soon update the PR according to the changes requested.

**Comment by sahilbansal17** [comment]
@aayushinigam Are you working on it?

**Comment by aayushinigam** [comment]
@sahilbansal17  No. Hence I closed it.

**Pull request closed without merging**